lib/actions/rootActions.js

```diff
@@ -232,11 +234,27 @@ export const setLoading = index => ({
   index
 });
 
-export const selectSavedQuery = (value, index) => ({
-  type: actions.SELECT_SAVED_QUERY,
-  value,
-  index
-});
+export const selectSavedQuery = (value, index) => (dispatch, getState) => {
+  const store = getState();
+  const dashboardInfo = getDashboardInfo(store);
+  const delValue = dashboardInfo.data.items[index].savedQuery;
+  const savedQueriesList = getSavedQueriesList(store);
+  dispatch(changeSavedQueryList(delValue, value));
+  const newStore = getState();
+  const newSavedQueriesList = getSavedQueriesList(newStore);
+  value.length > 0
+    ? value.some(sq => !savedQueriesList.includes(sq)) &&
```

Comment:
I would also suggest to split that conditions in and re-naming some variables to more readable form.
`
export const selectSavedQueries = (queryNames, idx) => (dispatch, getState)
...
const isPublic = dashboardInfo.is_public;

if (isPublic && queryNames.length) {
  const  isNewQueryAdded = queryNames.some(query=> !savedQueriesList.includes(query);
  ... 
}

`